File: static/js/students.js
from flask import request, jsonify, render_template, session
import db

# 📌 صفحة واجهة الطلاب (HTML + JS)
def students():
    return render_template("students.html")

# 📌 جلب الطلاب (مع البحث + الصف والشعبة)
def get_students():
    search = request.args.get("search", "").strip()
    conn = db.get_db_connection()

    if search:
        rows = conn.execute("""
            SELECT s.id, s.student_name, tc.class_name, tc.section
            FROM students s
            LEFT JOIN teacher_classes tc ON s.class_id = tc.id
            WHERE s.school_id = ? AND s.student_name LIKE ?
            ORDER BY s.id DESC
        """, (session["school_id"], f"%{search}%")).fetchall()
    else:
        rows = conn.execute("""
            SELECT s.id, s.student_name, tc.class_name, tc.section
            FROM students s
            LEFT JOIN teacher_classes tc ON s.class_id = tc.id
            WHERE s.school_id = ?
            ORDER BY s.id DESC
        """, (session["school_id"],)).fetchall()

    conn.close()
    return jsonify([dict(r) for r in rows])

# 📌 إضافة طالب
def add_student():
    data = request.get_json()
    name = data.get("student_name")
    class_name = data.get("class_name")
    section = data.get("section")

    if not name or not class_name or not section:
        return jsonify({"error": "⚠️ يرجى إدخال جميع البيانات"}), 400

    conn = db.get_db_connection()

    # التحقق من وجود الصف والشعبة
    class_row = conn.execute("""
        SELECT id FROM teacher_classes
        WHERE class_name = ? AND section = ? AND school_id = ?
    """, (class_name, section, session["school_id"])).fetchone()

    if class_row:
        class_id = class_row["id"]
    else:
        cur = conn.execute("""
            INSERT INTO teacher_classes (class_name, section, period, school_id)
            VALUES (?, ?, 'صباحي', ?)
        """, (class_name, section, session["school_id"]))
        class_id = cur.lastrowid

    conn.execute("""
        INSERT INTO students (student_name, class_id, school_id)
        VALUES (?, ?, ?)
    """, (name, class_id, session["school_id"]))

    conn.commit()
    conn.close()
    return jsonify({"success": True})

# 📌 تعديل طالب
def edit_student(student_id):
    data = request.get_json()
    new_name = data.get("student_name")
    new_class = data.get("class_name")
    new_section = data.get("section")

    if not new_name or not new_class or not new_section:
        return jsonify({"error": "⚠️ يرجى إدخال جميع البيانات"}), 400

    conn = db.get_db_connection()

    # التحقق من وجود الصف والشعبة الجديدة
    class_row = conn.execute("""
        SELECT id FROM teacher_classes
        WHERE class_name = ? AND section = ? AND school_id = ?
    """, (new_class, new_section, session["school_id"])).fetchone()

    if class_row:
        class_id = class_row["id"]
    else:
        cur = conn.execute("""
            INSERT INTO teacher_classes (class_name, section, period, school_id)
            VALUES (?, ?, 'صباحي', ?)
        """, (new_class, new_section, session["school_id"]))
        class_id = cur.lastrowid

    conn.execute("""
        UPDATE students
        SET student_name = ?, class_id = ?
        WHERE id = ?
    """, (new_name, class_id, student_id))

    conn.commit()
    conn.close()
    return jsonify({"success": True})

# 📌 حذف طالب
def delete_student(student_id):
    conn = db.get_db_connection()
    conn.execute("DELETE FROM students WHERE id = ?", (student_id,))
    conn.commit()
    conn.close()
    return jsonify({"success": True})

File: students.py
# students.py
from flask import render_template, request, redirect, url_for, jsonify, session, flash
from db import SessionLocal  # ✅ استدعاء جلسة SQLAlchemy
from auth import login_required

@login_required
def students():
    db_session = SessionLocal()
    school_id = session.get("school_id")

    if not school_id:
        flash("⚠️ لم يتم تحديد المدرسة")
        db_session.close()
        return redirect(url_for("dashboard"))

    if request.method == 'POST':
        # الحقل في القالب اسمه student_names (يمكن أن يكون عدة أسطر)
        student_names = request.form.get('student_names') or request.form.get('student_name', '')
        class_name = request.form.get('class_name', '')
        section = request.form.get('section', '')

        # دعم إدخال أسماء متعددة (كل سطر اسم أو أكثر)
        for name in student_names.splitlines():
            name = name.strip()
            if name:
                db_session.execute(
                    'INSERT INTO students (student_name, class_name, section, school_id) '
                    'VALUES (:student_name, :class_name, :section, :school_id)',
                    {"student_name": name, "class_name": class_name, "section": section, "school_id": school_id}
                )
        db_session.commit()
        db_session.close()
        return redirect(url_for('students'))

    # GET → إظهار طلاب المدرسة الحالية فقط
    students_list = db_session.execute(
        'SELECT * FROM students WHERE school_id = :school_id ORDER BY student_name',
        {"school_id": school_id}
    ).fetchall()
    db_session.close()
    return render_template('students.html', students=students_list)

@login_required
def edit_student(student_id):
    data = request.get_json()
    student_name = data.get('student_name')
    class_name = data.get('class_name')
    section = data.get('section')

    db_session = SessionLocal()
    # تحديث بشرط school_id من السيشن
    db_session.execute('''
        UPDATE students
        SET student_name=:student_name, class_name=:class_name, section=:section
        WHERE id=:student_id AND school_id=:school_id
    ''', {
        "student_name": student_name,
        "class_name": class_name,
        "section": section,
        "student_id": student_id,
        "school_id": session.get("school_id")
    })
    db_session.commit()
    db_session.close()
    return jsonify({'success': True})

@login_required
def delete_student(student_id):
    db_session = SessionLocal()
    # حذف بشرط school_id كمان
    db_session.execute('DELETE FROM students WHERE id=:student_id AND school_id=:school_id',
                       {"student_id": student_id, "school_id": session.get("school_id")})
    db_session.commit()
    db_session.close()
    return jsonify({'success': True})
